fastlane/lanes/localization.rb

```diff
@@ -122,49 +127,78 @@
   # @called_by complete_code_freeze
   #
   lane :generate_strings_file_for_glotpress do |options|
-    cocoapods
-
-    # On top of fetching the latest Pods, we also need to fetch the source for the Gutenberg code.
-    # To get it, we need to manually clone the repo, since Gutenberg is distributed via XCFramework.
-    # XCFrameworks are binary targets and cannot extract strings via genstrings from there.
-    config = gutenberg_config!
-
-    ref_node = config[:ref]
-    UI.user_error!('Could not find Gutenberg ref to clone the repository in order to access its strings.') if ref_node.nil?
-
-    ref = ref_node[:tag] || ref_node[:commit]
-    UI.user_error!('The ref to clone Gutenberg in order to access its strings has neither tag nor commit values.') if ref.nil?
-
-    github_org = config[:github_org]
-    UI.user_error!('Could not find GitHub organization name to clone Gutenberg in order to access its strings.') if github_org.nil?
-
-    repo_name = config[:repo_name]
-    UI.user_error!('Could not find GitHub repository name to clone Gutenberg in order to access its strings.') if repo_name.nil?
-
-    # Create a temporary directory to clone Gutenberg into.
-    # We'll run the rest of the automation from within the block, but notice that only the Gutenbreg cloning happens within the temporary directory.
-    gutenberg_clone_name = 'Gutenberg-Strings-Clone'
+    # We jump in the tempdir immediately, even though we might not need it.
+    # It's just simpler to rely on the Ruby block API for it than to manage it by hand.
     Dir.mktmpdir do |tempdir|
-      Dir.chdir(tempdir) do
-        repo_url = "https://github.com/#{github_org}/#{repo_name}"
-        UI.message("Cloning Gutenberg from #{repo_url} into #{gutenberg_clone_name}. This might take a few minutes…")
-        sh("git clone --depth 1 #{repo_url} #{gutenberg_clone_name}")
-        Dir.chdir(gutenberg_clone_name) do
-          if config[:ref][:tag]
-            sh("git fetch origin refs/tags/#{ref}:refs/tags/#{ref}")
-            sh("git checkout refs/tags/#{ref}")
-          else
-            sh("git fetch origin #{ref}")
-            sh("git checkout #{ref}")
+      # Fetch fresh pods to read the latest localizations from them.
+      # In CI, we expect the pods to be already available and up to date.
+      cocoapods unless is_ci
+
+      # For the same reason, fetch fresh packages
+      # FIXME: We yet don't have an explicit way to set the derived data in CI, so we'll run it in CI, too
+      derived_data_path = options[:derived_data_path] || DERIVED_DATA_PATH
+      resolve_packages(derived_data_path: derived_data_path)
```

Comment:
> I made the derived data path a parameter with default because I find passing a value makes the localization code a bit clearer, see below.

This is what I was referring to. I find using a dedicated value, `derived_data_path` between here and the paths list below makes the relationship between the two clear.